<img src="images/logo.jpg" style="display: block; margin-left: auto; margin-right: auto;" alt="לוגו של מיזם לימוד הפייתון. נחש מצויר בצבעי צהוב וכחול, הנע בין האותיות של שם הקורס: לומדים פייתון. הסלוגן המופיע מעל לשם הקורס הוא מיזם חינמי ללימוד תכנות בעברית.">

# <p style="text-align: right; direction: rtl; float: right;">תנאים – חלק 2</p>

## <p style="text-align: right; direction: rtl; float: right;">תנאים – תזכורת</p>

<p style="text-align: right; direction: rtl; float: right;">
ניזכר במחברת הקודמת, שבה למדנו על תנאים.<br>
למדנו שבעזרת מילת המפתח <code>if</code> אנחנו יכולים לבקש מהקוד שלנו לבצע סדרת פעולות, רק אם תנאי כלשהו מתקיים.<br>
במילים אחרות: אנחנו יכולים לבקש מקטע קוד לרוץ, רק אם ביטוי בוליאני מסוים שווה ל־<code>True</code>.
</p>

<p style="text-align: right; direction: rtl; float: right;">
נראה דוגמה:
</p>

In [ ]:
shoes_in_my_drawer = int(input("How many shoes do you have in your drawer? "))

ifshoes_in_my_drawer % 2 == 0:
    print?("everything is good")

if shoes_in_my_drawer % 2 == 1:
    print("You have an odd number of shoes. Something is wrong!")

<p style="text-align: right; direction: rtl; float: right;">
בקוד שלמעלה, ביקשנו מהמשתמש להזין את מספר הנעליים שיש לו, ואם המספר היה אי־זוגי הדפסנו לו שמשהו מוזר מתרחש.<br>
אבל מה קורה אם נרצה להדפיס למשתמש הודעת אישור כשאנחנו מגלים שהכול בסדר?
</p>

<div class="align-center" style="display: flex; text-align: right; direction: rtl; clear: both;">
    <div style="display: flex; width: 10%; float: right; clear: both;">
        <img src="images/exercise.svg" style="height: 50px !important;" alt="תרגול"> 
    </div>
    <div style="width: 70%">
        <p style="text-align: right; direction: rtl; float: right; clear: both;">
            כיצד הייתם משפרים את התוכנית שלמעלה כך שתדפיס למשתמש הודעת אישור שהכול בסדר?<br> 
            השתמשו בכלים שרכשתם במחברת הקודמת. נסו לחשוב על לפחות 2 דרכים דומות.
        </p>
    </div>
    <div style="display: flex; width: 20%; border-right: 0.1rem solid #A5A5A5; padding: 1rem 2rem;">
        <p style="text-align: center; direction: rtl; justify-content: center; align-items: center; clear: both;">
            <strong>חשוב!</strong><br>
            פתרו לפני שתמשיכו!
        </p>
    </div>
</div>

## <span style="text-align: right; direction: rtl; float: right;">מה אם לא?<br><sub>על מילת המפתח else</sub></span>

<p style="text-align: right; direction: rtl; float: right;">דרך אחת לפתור את התרגיל שהופיע למעלה, היא זו:</p>

In [1]:
shoes_in_my_drawer = int(input("How many shoes do you have in your drawer? "))

if shoes_in_my_drawer % 2 == 1:
    print("You have an odd number of shoes. Something is wrong!")
if shoes_in_my_drawer % 2 != 1:
    print("You have an even number of shoes. Congratulations!")

How many shoes do you have in your drawer? 3
You have an odd number of shoes. Something is wrong!


<p style="text-align: right; direction: rtl; float: right;">
כדי להדפיס למשתמש הודעה מתאימה בכל מצב, הוספנו תנאי <em>הפוך</em> מהתנאי הראשון, שידפיס למשתמש הודעה מתאימה.<br>
אחד הדברים המעניינים בתוכנית שלמעלה, הוא שיש לנו שני תנאים שמנוגדים זה לזה במשמעות שלהם. אחד בודק זוגיות, והשני בודק אי־זוגיות.<br>
למעשה, אם היינו רוצים לנסח את הקוד במילים, היינו יכולים להגיד: <q>אם מספר הנעליים הוא אי־זוגי, הדפס שיש בעיה. <strong>אחרת</strong>, הדפס שהכול בסדר</q>.
</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
    פייתון מקנה לנו כלי נוח לבטא את ה"אחרת" הזה, כלי שמקל על הקריאוּת של הקוד – מילת המפתח <code>else</code>.<br>
    נראה איך אפשר לנסח את הקוד שלמעלה בעזרת <code>else</code>:
</p>

In [ ]:
shoes_in_my_drawer = int(input("How many shoes do you have in your drawer? "))

if shoes_in_my_drawer % 2 == 1:
    print("You have an odd number of shoes. Something is wrong!")
else:
    print("You have an even number of shoes. Congratulations!")

<p style="text-align: right; direction: rtl; float: right; clear: both;">
שימו לב לצורת ההזחה: ה־<code>else</code> אינו מוזח, אך התוכן שבתוכו כן.<br>
נזכור גם ש־<code>else</code> קשור ל־<code>if</code> שלפניו, ומדבר על <em>המקרה המשלים</em> לתנאי שנמצא באותו <code>if</code>.<br>
דרך נוספת לחשוב על <code>else</code> היא ש<mark>קטע הקוד בתוך ה־<code>else</code> יתבצע אם תוצאתו של הביטוי הבוליאני שמופיע כתנאי של ה־<code>if</code> היא <code>False</code></mark>.
</p>

### <div style="text-align: right; direction: rtl; float: right; clear: both;">דוגמאות לתנאים עם <code>else</code></div>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בדוגמאות הבאות, <em style='background-color: #3C6478; color: white;'>התנאי מופיע כך</em>, ו<em style='background-color: #F58B4C; color: white;'>הפעולות שיקרו בעקבותיו מופיעות כך</em>.
</p>

<ol style="text-align: right; direction: rtl; float: right; clear: both;">
<li>אם <em style='background-color: #3C6478; color: white;'>השעה היא לפני 20:00</em>, <em style='background-color: #F58B4C; color: white;'>שמור על תאורה גבוהה במסעדה</em>. <em style='background-color: #3C6478; color: white;'>אחרת</em>, <em style='background-color: #F58B4C; color: white;'>עמעם את התאורה</em>.</li>
<li>אם <em style='background-color: #3C6478; color: white;'>הגיל של המשתמש הוא לפחות 18</em>, <em style='background-color: #F58B4C; color: white;'>אפשר לו להיכנס לבר</em>. <em style='background-color: #3C6478; color: white;'>אחרת</em>, <em style='background-color: #F58B4C; color: white;'>הצע לו אטרקציות אחרות לבקר בהן</em> וגם <em style='background-color: #F58B4C; color: white;'>אל תכניס אותו</em>.</li>
<li>אם המשתמש בכספומט <em style='background-color: #3C6478; color: white;'>הזין סכום שלילי, או יותר מהסכום הזמין בחשבון שלו</em>, <em style='background-color: #F58B4C; color: white;'>הצג לו הודעת שגיאה</em>. <em style='background-color: #3C6478; color: white;'>אחרת</em>, <em style='background-color: #F58B4C; color: white;'>הפחת לו את הסכום מהחשבון</em>, <em style='background-color: #F58B4C; color: white;'>הוצא לו שטרות בסכום שהזין</em>.</li>
<li>אם <em style='background-color: #3C6478; color: white;'>הדוד כבוי וגם השעה היא לפני 8:00 וגם השעה היא אחרי 7:00</em>, <em style='background-color: #F58B4C; color: white;'>הדלק את הדוד</em>. <em style='background-color: #3C6478; color: white;'>אחרת</em>, אם <em style='background-color: #3C6478; color: white;'>הדוד דלוק</em> – <em style='background-color: #F58B4C; color: white;'>כבה את הדוד</em>.</li>
</ol>

### <div style="text-align: right; direction: rtl;">זרימת התוכנית: ציור לדוגמה</div>

<figure>
    <img src="images/else-flow.svg" style="display: block; margin-left: auto; margin-right: auto;" alt="איור (גרף) להמחשה של זרימת התוכנה. מתחילים בלקבל קלט, עוברים לתנאי. אם הוא לא מתקיים, בודקים אם הדוד דלוק. אם הדוד דלוק, מכבים את הדוד ומשם עוברים לסוף התוכנית. אם התנאי ההתחלתי קטן מתקיים, מדליקים את הדוד, ואז מגיעים לסוף התוכנית.">
    <figcaption style="text-align: center; direction: rtl;"></figcaption>
</figure>

### <span style="text-align: right; direction: rtl; float: right; clear: both;">מימוש לדוגמה</span>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
התוכנית המוצגת פה מעט מורכבת יותר מהשרטוט המופיע למעלה, כדי לתת לה נופך מציאותי יותר.
</p>

In [2]:
is_boiler_on = input("Is your boiler turned on? [yes/no] ")
hour = int(input("Enter the hour (00-23): "))
minute = int(input("Enter the minute (00-59): "))

if is_boiler_on == 'yes':
    is_boiler_on = True
else:
    is_boiler_on = False

if not is_boiler_on and hour == 7 and minute > 0:
    is_boiler_on = True
else:
    if is_boiler_on:
        is_boiler_on = False

if is_boiler_on:
    boiler_status = "on"
else:
    boiler_status = "off"

print("Boiler is " + boiler_status + " right now.")

Is your boiler turned on? [yes/no] no
Enter the hour (00-23): 7
Enter the minute (00-59): 0
Boiler is off right now.


## <span style="text-align: right; direction: rtl; float: right; clear: both;">מה המצב?<br><sub>טיפול במצבים מרובים</sub></span>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
בארצות הברית מדורגת מכירת אלבומים כך:
</p>
<ol style="text-align: right; direction: rtl; float: right; clear: both;">
    <li>אלבום מוזיקה נחשב "<dfn>אלבום כסף</dfn>" אם נמכרו ממנו לפחות <em>100,000</em> עותקים.</li>
    <li>אלבום מוזיקה נחשב "<dfn>אלבום זהב</dfn>" אם נמכרו ממנו לפחות <em>500,000</em> עותקים.</li>
    <li>אלבום מוזיקה נחשב "<dfn>אלבום פלטינה</dfn>" אם נמכרו ממנו לפחות <em>1,000,000</em> עותקים.</li>
    <li>אלבום מוזיקה נחשב "<dfn>אלבום יהלום</dfn>" אם נמכרו ממנו לפחות <em>10,000,000</em> עותקים.</li>
</ol>

In [3]:
num_of_sold_copies = int(input("Enter the number of sold copies of a certain album:"))

if num_of_sold_copies >= 10000000 :
    print("your album is ranked as diamind album")
elif num_of_sold_copies >= 1000000 :
    print("your album is ranked as platine album")
elif num_of_sold_copies >= 500000 :
    print("your album is ranked as gold album")    
elif num_of_sold_copies >= 100000 :
    print("your album is ranked as silver album")

else:
    print("this album is not best seller")

Enter the number of sold copies of a certain album:520196
your album is ranked as gold album


<div class="align-center" style="display: flex; text-align: right; direction: rtl; clear: both;">
    <div style="display: flex; width: 10%; float: right; clear: both;">
        <img src="images/exercise.svg" style="height: 50px !important;" alt="תרגול"> 
    </div>
    <div style="width: 70%">
        <p style="text-align: right; direction: rtl; float: right; clear: both;">
            קלטו את כמות העותקים שנמכרו עבור אלבום המטאל המצליח "רוצח או נזלת", והדפיסו את דירוג האלבום.<br>
            לדוגמה, אם המשתמש הכניס שמספר המכירות היה <em>520,196</em>, הדפיסו "<samp>אלבום זהב</samp>".<br>
            אם האלבום לא נמכר מספיק כדי לזכות בדירוג, הדפיסו "<samp>האלבום אינו רב־מכר</samp>".<br>
        </p>
    </div>
    <div style="display: flex; width: 20%; border-right: 0.1rem solid #A5A5A5; padding: 1rem 2rem;">
        <p style="text-align: center; direction: rtl; justify-content: center; align-items: center; clear: both;">
            <strong>חשוב!</strong><br>
            פתרו לפני שתמשיכו!
        </p>
    </div>
</div>

### <span style="text-align: right; direction: rtl; float: right; clear: both;">פתרון ושיפורים</span>

<p style="text-align: right; direction: rtl; float: right;">
בתרגיל שנתבקשתם לפתור, נוצר מצב שבו יש צורך בתנאים מרובים שמשלימים זה את זה.<br>
נציג לפניכם שני פתרונות אפשריים:
</p>

In [4]:
copies_sold = int(input("How many copies were sold? "))

# אנחנו משתמשים באותיות גדולות עבור שמות המשתנים האלו.
# זו מוסכמה בין מתכנתים שמבקשת להדגיש שערכי המשתנים הם קבועים, ולא הולכים להשתנות במהלך התוכנית.
SILVER_ALBUM = 100000
GOLD_ALBUM = 500000
PLATINUM_ALBUM = 1000000
DIAMOND_ALBUM = 10000000


if copies_sold >= DIAMOND_ALBUM:
    print("Diamond album")
else:
    if copies_sold >= PLATINUM_ALBUM:
        print("Platinum album")
    else:
        if copies_sold >= GOLD_ALBUM:
            print("Gold album")
        else:
            if copies_sold >= SILVER_ALBUM:
                print("Silver album")
            else:
                print("Your album is not a best-seller")

How many copies were sold? 520000
Gold album


<p style="text-align: right; direction: rtl; float: right;">
ודאי שמתם לב שהקוד נראה מעט מסורבל בגלל כמות ההזחות, והוא יוסיף ויסתרבל ככל שנוסיף יותר מקרים אפשריים.<br>
ננסה לפתור את זה בעזרת הגדרת טווחים מדויקים עבור כל דירוג.<br>
הקוד המשופץ ייראה כך:
</p>

In [8]:
copies_sold = int(input("How many copies were sold? "))

SILVER_ALBUM = 100000
GOLD_ALBUM = 500000
PLATINUM_ALBUM = 1000000
DIAMOND_ALBUM = 10000000

if copies_sold >= DIAMOND_ALBUM:
    print("Diamond album")
if copies_sold >= PLATINUM_ALBUM and copies_sold < DIAMOND_ALBUM:
    print("Platinum album")
if copies_sold >= GOLD_ALBUM and copies_sold < PLATINUM_ALBUM:
    print("Gold album")
if copies_sold >= SILVER_ALBUM and copies_sold < GOLD_ALBUM:
    print("Silver album")
if copies_sold < SILVER_ALBUM:
    print("Your album is not a best-seller")

How many copies were sold? 100000000
Diamond album
Platinum album
Gold album
Silver album


<p style="text-align: right; direction: rtl; float: right;">
הקוד נראה טוב יותר במידה ניכרת, אבל בכל <code>if</code> אנחנו בודקים שהתנאי שלפניו לא התקיים, וזה מסורבל מאוד.<br>
אנחנו עושים את זה כדי למנוע הדפסה כפולה: אומנם כל אלבום זהב נמכר מספיק פעמים כדי להיות מוכתר כאלבום כסף, אבל לא נרצה להדפיס למשתמש את שתי ההודעות.</p>

<div class="align-center" style="display: flex; text-align: right; direction: rtl; clear: both;">
    <div style="display: flex; width: 10%; float: right; clear: both;">
        <img src="images/exercise.svg" style="height: 50px !important;" alt="תרגול"> 
    </div>
    <div style="width: 70%">
        <p style="text-align: right; direction: rtl; float: right; clear: both;">
            מה היה קורה אם לא היו תנאים אחרי האופרטור הלוגי <code>and</code>?<br>
            מחקו אותם, הכניסו לתוכנה כקלט <em>10000000</em> ובדקו מה התוצאה.
        </p>
    </div>
    <div style="display: flex; width: 20%; border-right: 0.1rem solid #A5A5A5; padding: 1rem 2rem;">
        <p style="text-align: center; direction: rtl; justify-content: center; align-items: center; clear: both;">
            <strong>חשוב!</strong><br>
            פתרו לפני שתמשיכו!
        </p>
    </div>
</div>

### <div style="text-align: right; direction: rtl; float: right; clear: both;">אם אחרת – elif</div>

<p style="text-align: right; direction: rtl; float: right;">
כדי לפתור את הבעיה שהוצגה למעלה, פייתון מעניקה לנו כלי נוסף שנקרא <code>elif</code>.<br>
מדובר בסך הכול בקיצור של <code dir="ltr" style="direction: ltr">else... if (תנאי)</code>, או בעברית: אם התנאי הקודם לא התקיים, בדוק אם...<br>
ראו, לדוגמה, איך נשפר את הקוד הקודם לקוד קריא יותר <em>במידה רבה</em>:
</p>

In [ ]:
copies_sold = int(input("How many copies were sold? "))

SILVER_ALBUM = 100000
GOLD_ALBUM = 500000
PLATINUM_ALBUM = 1000000
DIAMOND_ALBUM = 10000000

if copies_sold >= DIAMOND_ALBUM:
    print("Diamond album")
elif copies_sold >= PLATINUM_ALBUM:
    print("Platinum album")
elif copies_sold >= GOLD_ALBUM:
    print("Gold album")
elif copies_sold >= SILVER_ALBUM:
    print("Silver album")
else:
    print("Your album is not a best-seller")

### <div style="text-align: right; direction: rtl; float: right; clear: both;">מה קורה כאן?</div>
<p style="text-align: right; direction: rtl; float: right; clear: both;">
    הטריק הוא שבשורה שבה כתוב <code>elif</code>, פייתון תנסה לבדוק את התנאי <em>רק אם</em> התנאים שלפניו לא התקיימו.<br>
    במילים אחרות – ערכיהם של הביטויים הבוליאניים בכל התנאים שלפניו היו <code>False</code>.<br>
    בכל שורה שבה יש <code>if</code> או <code>elif</code>, פייתון בודקת האם הביטוי הבוליאני שבאותה שורה מתקיים, ואז:
</p>

<ol style="text-align: right; direction: rtl; float: right; clear: both;">
    <li>אם כן, היא מבצעת את הפעולות המוזחות מתחת לתנאי ומפסיקה לבדוק את התנאים הבאים.</li>
    <li>אם לא, היא עוברת לבדוק את התנאי ב־<code>elif</code>־ים הבאים (אם ישנם <code>elif</code>־ים).</li>
    <li>אם אף אחד מהתנאים לא מתקיים, יתבצע קטע הקוד ששייך ל־<code>else</code> (אם יש <code>else</code>).</li>
</ol>

<div class="align-center" style="display: flex; text-align: right; direction: rtl;">
    <div style="display: flex; width: 10%; float: right; ">
        <img src="images/warning.png" style="height: 50px !important;" alt="אזהרה!"> 
    </div>
    <div style="width: 90%">
        <p style="text-align: right; direction: rtl;">
            ניתן לכתוב <code>if</code> בלי <code>else</code> ובלי <code>elif</code>־ים אחריו.<br>
            <code>if</code> תמיד יהיה ראשון, ואם יש צורך להוסיף מקרים, <code>else</code> תמיד יהיה אחרון, וביניהם יהיו <code>elif</code>־ים.<br>
        </p>
    </div>
</div>

## <p style="align: right; direction: rtl; float: right;">תרגול</p>

### <p style="align: right; direction: rtl; float: right;">כניסה לבנק, שלב 2</p>

<p style="text-align: right; direction: rtl; float: right;">
שם המשתמש שלי לבנק הוא <em>wrong</em>, והסיסמה שלי היא <em>ads sports</em>.<br>
שם המשתמש של מנהל הבנק היא <em>admin</em>, והסיסמה שלו היא <em>is trator</em>.
<br>
קבלו מהמשתמש שם משתמש וסיסמה, והדפיסו לו הודעה שמספרת לו לאיזה משתמש הוא הצליח להתחבר.<br>
אם לא הצליח להתחבר, הדפיסו לו הודעת שגיאה מתאימה.
</p>

In [9]:
#upload 15
my_user_name = "wrong"

manager_user_name = "admin"

my_password = "ads sports"

manager_password = "is trator"

input_user_name = input("Enter your user name:")

input_password = input("Enter your password:")

if input_user_name == my_user_name and input_password == my_password:
    print("Hello Hagai, welcome to your account!")

elif input_user_name == manager_user_name and input_password == manager_password:
    print("Hello bank manager, how can I help you today?")

else:
    print("I'm sorry password or/and user name are not correct ")

Enter your user name:admin
Enter your password:is trator
Hello bank manager, how can I help you today?


### <p style="align: right; direction: rtl; float: right;">חשבון למתחילים, שלב 1</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
דני רוצה ללמוד חשבון, ולצורך כך הוא צריך מחשבון פשוט שיעזור לו.<br>
כתבו תוכנה שמקבלת 2 מספרים ופעולה חשבונית (<em>+</em>, <em>-</em>, <em>*</em>, <em>/</em> או <em>**</em>), ויודעת להחזיר את התשובה הנכונה לתרגיל.<br>
</p>
<p style="text-align: right; direction: rtl; float: right; clear: both;">
לדוגמה:
</p>
<ul style="text-align: right; direction: rtl; float: right; clear: both; margin-top: 0;">
    <li>עבור מספר ראשון <em>5</em>, מספר שני <em>2</em> ופעולה <em>/</em> התוכנית תדפיס <samp>2.5</samp>, כיוון ש־<span dir="ltr">5/2 == 2.5</span>.</li>
    <li>עבור מספר ראשון <em>9</em>, מספר שני <em>2</em> ופעולה <em>**</em> התוכנית תדפיס <samp>81</samp>, כיוון ש־<span dir="ltr">9<sup>2</sup> == 81</span>.</li>
    <li>עבור מספר ראשון <em>3</em>, מספר שני <em>7</em> ופעולה <em>-</em> התוכנית תדפיס <samp dir="ltr" style="direction: ltr;">-4</samp>, כיוון ש־<span dir="ltr">3-7 == -4</span>.</li>
</ul>

In [3]:
#upload 16

def convert_to_num(param_str):
    
    if "." in param_str:
        num = float(param_str)
    else:
        num = int(param_str)
    
    return num    

num1 = input("Enter the first number:")
num2 = input("Enter the second number:")

num1 = convert_to_num(num1)
num2 = convert_to_num(num2)


math_operation = input("Enter the math operation you would like to use, + , - , * , / , **: ")

result = 0

if math_operation == '+':
    result = num1 + num2

elif math_operation == '-':
    result = num1 - num2

elif math_operation == '*':
    result = num1 * num2

elif math_operation == '/':
    result = num1 / num2

elif math_operation == '**':
    result = num1 ** num2

else:
    print("You have entered unrecognized charachter for math operations.")

print(result)

Enter the first number:2.4
Enter the second number:1.8
Enter the math operation you would like to use, + , - , * , / , **: *
4.32


### <p style="align: right; direction: rtl; float: right;">מחשבון מס הכנסה</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
    המיסוי על הכנסה בישראל גבוה מאוד ושיעורו נקבע לפי מדרגות כדלקמן:<br>
</p>
<ol style="text-align: right; direction: rtl; float: right; clear: both;">
    <li>מי שמרוויח עד 6,310 ש"ח, משלם מס בשיעור 10% על הסכום הזה.</li>
    <li>מי שמרוויח עד 9,050 ש"ח, משלם מס בשיעור 14% על הסכום הזה.</li>
    <li>מי שמרוויח עד 14,530 ש"ח, משלם מס בשיעור 20% על הסכום הזה.</li>
    <li>מי שמרוויח עד 20,200 ש"ח, משלם מס בשיעור 31% על הסכום הזה.</li>
    <li>מי שמרוויח עד 42,030 ש"ח, משלם מס בשיעור 35% על הסכום הזה.</li>
    <li>מי שמרוויח עד 54,130 ש"ח, משלם מס בשיעור 47% על הסכום הזה.</li>
    <li>מי שמרוויח מעל הסכום האחרון, משלם מס בשיעור 50% על הסכום הזה.</li>
</ol>
<p style="text-align: right; direction: rtl; float: right; clear: both;">
    הסכום תמיד משולם על ההפרש בין המדרגות.<br>
    לדוגמה, אם הרווחתי בחודש מסוים 10,000 ש"ח, תשלום המיסים שלי יחושב כך:<br>
    על 6,310 השקלים הראשונים אשלם 631 ש"ח, שהם 10% מאותה מדרגה.<br>
    על הסכום העודף עד 9,050 שקלים, שהוא 2,740 שקלים <span dir="ltr" style="direction: ltr">(9,050 - 6,310)</span> אשלם 383.6 שקלים, שהם 14% מ־2,740 שקלים.<br>
    על הסכום העודף בסך 950 שקלים <span dir="ltr" style="direction: ltr">(10,000 - 9,050)</span> אשלם 190 ש"ח, שהם 20% מ־950 שקלים.<br>
    בסך הכול, אשלם למס הכנסה באותו חודש 631 + 383.6 + 190 ש"ח, שהם 1,204.6 שקלים.
</p>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
    כתבו קוד למחשבון שמקבל את השכר החודשי שלכם ומדפיס את סכום המס שתשלמו.
<p>

In [1]:
#upload 17
STEP_1_SALARY = 6310
TAX_PERCENT_FOR_STEP_1 = 0.1
FULL_TAXES_FOR_STEP_1 = STEP_1_SALARY * TAX_PERCENT_FOR_STEP_1  # = 631

STEP_2_SALARY = 9050
TAX_PERCENT_FOR_STEP_2 = 0.14
FULL_TAXES_FOR_STEP_2 = (STEP_2_SALARY - STEP_1_SALARY) * TAX_PERCENT_FOR_STEP_2  # = 383.6

STEP_3_SALARY = 14530
TAX_PERCENT_FOR_STEP_3 = 0.2
FULL_TAXES_FOR_STEP_3 = (STEP_3_SALARY - STEP_2_SALARY) * TAX_PERCENT_FOR_STEP_3  # = 1096

STEP_4_SALARY = 20200
TAX_PERCENT_FOR_STEP_4 = 0.31
FULL_TAXES_FOR_STEP_4 = (STEP_4_SALARY - STEP_3_SALARY) * TAX_PERCENT_FOR_STEP_4  # = 1757.7

STEP_5_SALARY = 42030
TAX_PERCENT_FOR_STEP_5 = 0.35
FULL_TAXES_FOR_STEP_5 = (STEP_5_SALARY - STEP_4_SALARY) * TAX_PERCENT_FOR_STEP_5  # = 7640.5

STEP_6_SALARY = 54130
TAX_PERCENT_FOR_STEP_6 = 0.47
FULL_TAXES_FOR_STEP_6 = (STEP_6_SALARY - STEP_5_SALARY) * TAX_PERCENT_FOR_STEP_6  # = 5687

TAX_PERCENT_FOR_STEP_7 = 0.5

salary = float(input("Enter your salary:"))

accumulated_taxes = 0
deficit_between_salary_and_former_step_ = 0

if salary <= STEP_1_SALARY:
    accumulated_taxes = salary * TAX_PERCENT_FOR_STEP_1

elif STEP_1_SALARY < salary < STEP_2_SALARY:
    deficit_between_salary_and_former_step = salary - STEP_1_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_2
    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_1

elif STEP_2_SALARY < salary < STEP_3_SALARY:
    deficit_between_salary_and_former_step = salary - STEP_2_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_3
    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_2 + FULL_TAXES_FOR_STEP_1

elif STEP_3_SALARY < salary < STEP_4_SALARY:
    deficit_between_salary_and_former_step = salary - STEP_3_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_4
    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_3 + FULL_TAXES_FOR_STEP_2 + FULL_TAXES_FOR_STEP_1

elif STEP_4_SALARY < salary < STEP_5_SALARY:
    deficit_between_salary_and_former_step = salary - STEP_4_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_5
    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_4 + FULL_TAXES_FOR_STEP_3 + FULL_TAXES_FOR_STEP_2 + FULL_TAXES_FOR_STEP_1

elif STEP_5_SALARY < salary < STEP_6_SALARY:
    deficit_between_salary_and_former_step = salary - STEP_5_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_6
    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_5 + FULL_TAXES_FOR_STEP_4 + FULL_TAXES_FOR_STEP_3 + FULL_TAXES_FOR_STEP_2 + FULL_TAXES_FOR_STEP_1

elif STEP_6_SALARY < salary:
    deficit_between_salary_and_former_step = salary - STEP_6_SALARY
    tax_from_last_step = deficit_between_salary_and_former_step * TAX_PERCENT_FOR_STEP_7
    print(deficit_between_salary_and_former_step)

    accumulated_taxes = tax_from_last_step + FULL_TAXES_FOR_STEP_6 + FULL_TAXES_FOR_STEP_5 + FULL_TAXES_FOR_STEP_4 + FULL_TAXES_FOR_STEP_3 + FULL_TAXES_FOR_STEP_2 + FULL_TAXES_FOR_STEP_1

print("The taxes you are paying for your salary is: " + str(accumulated_taxes) + " shekels.")    

Enter your salary:10000
The taxes you are paying for your salary is: 1204.6 shekels.


### <span style="align: right; direction: rtl; float: right; clear: both;">ירוץ אם נתקן, אחרת...</span>

<p style="text-align: right; direction: rtl; float: right; clear: both;">
    בקוד הבא נפלו שגיאות רבות.<br>
    תקנו אותו והריצו אותו כך שיעבוד ושידפיס הודעה אחת בלבד עבור כל מצב.
</p>

In [16]:
#upload 18
age = int(input("Please enter your age: "))

if age < 0:
    print("Your age is invalid.")
elif age < 18:
    print("Younger than 18.")
else:
    print("You are so old!")

Please enter your age: -90
Your age is invalid.
